sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample08_EventProcessor.md

```diff
@@ -0,0 +1,230 @@
+
+While you could write code directly using the `EventHubConsumerClient` to handle these cases, the Azure SDK also provides a type, `EventProcessorClient`, which handles this for you. It uses the Azure Blob Storage SDK to handle storing the state that's used during the load balancing and checkpointing operations. For many customers, that's sufficient, but some customers have asked for the ability to control how the information related to checkpointing and load balancing is stored. But for customers who have unique needs, we've exposed the lower-level machinery needed to build your own processor, which gives you full control over how this data is stored.
+
+## Building an Event Processor with `EventProcessor<TPartition>`
+
+The low level machinery needed to build you own processor, with full control over how its state is stored, is exposed by the `EventProcessor<TPartition>` class in the `Azure.Messaging.EventHubs.Primitives` namespace. This is an abstract class, which has five methods you'll need to provide implementations for. If you've used the `EventProcessorClient`, the first two will be familiar to you:
+
+- `OnProcessingEventBatchAsync`: The actual "business logic" of your processor. This is similar to the `ProcessEvent` event exposed by `EventProcessorClient`.
+- `OnProcessingErrorAsync`: A handler that can be used to observe exceptions from inside the machinery of the EventProcessor itself. This is similar to the `ProcessError` event exposed by `EventProcessorClient`.
+
+The remaining three deal with checkpointing and load balancing.
+
+- `ListOwnershipAsync`: Used by load balancing to see which partitions have been assigned to which processors.
+- `ClaimOwnershipAsync`: Used by load balancing to grab ownership of a partition (which may or may not be assigned to another processor).
+- `ListCheckpointsAsync`: Used to find the starting point inside a partition when a processor starts reading events from a partition.
+
+The two `On*` methods can only be implemented when we know what we want our event processor to do, but the remaining can be implemented independent of the business logic for processing events.
+
+In this sample, we will use the same storage mechanism as EventProcessorClient, i.e. we will store checkpoint and load balancing state using Azure Blob Storage.
+
+```C# Snippet:EventHubs_Sample08_AzureBlobStorageEventProcessor
```

Comment:
We probably want to give the line length a look here as well.   We've gotten feedback from customers letting us know that they like to read our samples directly on the GitHub site and asking that we be careful that the code lines aren't too long for that notoriously small area.